### PR TITLE
Documentation for new size unit system

### PR DESCRIPTION
**Pull request by Pessimistress**
For #6181 

#### Change List
- Documenting changes made in #6117 
- Upgrade guide

### CODE REVIEW

File: docs/api-reference/core/layer.md [hunk 1]
```diff
@@ -241,9 +241,9 @@ See the article on [Coordinate Systems](/docs/developer-guide/coordinate-systems
 
 ##### `coordinateOrigin` ([Number, Number], optional)
 
-Required when the `coordinateSystem` is set to `COORDINATE_SYSTEM.METER_OFFSETS`.
+Required if the `coordinateSystem` is set to offset modes.
```
Comment (kylebarron) [inline docs/api-reference/core/layer.md:244]:
```suggestion
Required if the `coordinateSystem` is set to one of the offset modes.
```
Maybe slightly clearer?

File: docs/developer-guide/coordinate-systems.md [hunk 2]
```diff
@@ -2,11 +2,9 @@
 
 In most deck.gl layers, every data object is expected to contain one or more **positions** (e.g. the center of a point, or the start and end of an arc, the vertices of a polygon, etc). Positions are expected to be supplied as two or three element arrays (`[x, y]` or `[x, y, z]`). Objects can also have **dimensions** (e.g. the radius of a circle, the width of a path, etc), most of which are specified as a single number. Positions and dimensions are used to construct a 3D world with your data.
 
-By default, deck.gl layers interprets positions as geospatial coordinates (i.e. `[longitude, latitude, altitude]`), and dimensions as in meters. So when working with geospatial data sources such as GeoJSON, deck.gl will automatically interpret any positions correctly.
+deck.gl supports a selection of coordinate systems, addressing diverse geospatial and non-geospatial [use cases](#world-space). Each layer can set its own `coordinateSystem` prop to specify how its positions should be interpreted. Layers using different coordinate systems can be composed together, which is very useful when dealing with datasets from different sources.
```
Comment (kylebarron) [inline docs/developer-guide/coordinate-systems.md:5]: I'm a little surprised the `use cases` link points to `#world-space`. Is that intentional?

File: docs/developer-guide/coordinate-systems.md [hunk 3]
```diff
@@ -19,42 +17,69 @@ new PointCloudLayer({
...
+Throughout this document you will find reference to the following **coordinate spaces** in deck.gl's projection system:
+
+##### World space
+
+The natural coordinate system of a dataset. It is usually determined by the source of the data, like the device it is generated on, and the format that it is stored as on disk or a cloud data warehouse.
+
+Some examples of different world space definitions include a [Shapefile](https://en.wikipedia.org/wiki/Shapefile) with any [standard geospatial reference](https://epsg.io/); WGS84 as used by GeoJSON or KML; GPS traces; LIDAR scans; info-vis values that do not map to real-world dimensions; [Slippy Map](https://wiki.openstreetmap.org/wiki/Slippy_Map) tiles from services like OpenStreetMaps and Mapbox; 3D tiling standards such as [Cesium 3D Tiles](https://www.ogc.org/standards/3DTiles) and [Esri I3S](https://www.ogc.org/standards/i3s).
+
+Positions in different world spaces come with different scales and orientations, may be measured relative to static or dynamic reference points, and in many cases are not linear. It is common for an application to overlay multiple datasets from different world spaces into the same 3D view.
```
Comment (kylebarron) [inline docs/developer-guide/coordinate-systems.md:36]:
```suggestion
Positions in different world spaces come with different scales and orientations, may be measured relative to static or dynamic reference points, and in many cases their scales are not linear. It is common for an application to overlay multiple datasets from different world spaces into the same 3D view.
```
Not sure what the best wording is, but the existing `in many cases are not linear` seemed to be referencing `Positions` which didn't seem right to me.

File: docs/developer-guide/coordinate-systems.md [hunk 4]
```diff
@@ -19,42 +17,69 @@ new PointCloudLayer({
...
+
+##### Common space
+
+To correctly compose data from various world spaces together, deck.gl transforms them into common space - an unified, intermediate 3D space that is a [right-handed Cartesian coordinate system](https://en.wikipedia.org/wiki/Cartesian_coordinate_system#In_three_dimensions). Once positions are in the common space, it is safe to add, substract, rotate, scale and extrude them as 3D vectors using standard linear algebra. This is the basis of all geometry processing in deck.gl layers.
```
Comment (kylebarron) [inline docs/developer-guide/coordinate-systems.md:40]:
```suggestion
To correctly compose data from various world spaces together, deck.gl transforms them into common space - a unified, intermediate 3D space that is a [right-handed Cartesian coordinate system](https://en.wikipedia.org/wiki/Cartesian_coordinate_system#In_three_dimensions). Once positions are in the common space, it is safe to add, subtract, rotate, scale and extrude them as 3D vectors using standard linear algebra. This is the basis of all geometry processing in deck.gl layers.
```

File: docs/developer-guide/coordinate-systems.md [hunk 5]
```diff
@@ -19,42 +17,69 @@ new PointCloudLayer({
...
+
+The transformation between the world space and the common space is referred to in deck.gl documentation as "project" and "unproject", a process controlled by both the specification of the world space, such as WGS84, and the [projection mode](/docs/developer-guide/views.md), such as Web Mercator. Projections are implemented as part of deck.gl's core.
```
Comment (kylebarron) [inline docs/developer-guide/coordinate-systems.md:42]:
```suggestion
The transformation between the world space and the common space is referred to in deck.gl documentation as "project" (world space to common space) and "unproject" (common space to world space), a process controlled by both the specification of the world space, such as WGS84, and the [projection mode](/docs/developer-guide/views.md), such as Web Mercator. Projections are implemented as part of deck.gl's core.
```
Not sure if it's helpful here to be more explicit about project/unproject. I always forget which is which.

File: docs/developer-guide/coordinate-systems.md [hunk 6]
```diff
@@ -19,42 +17,69 @@ new PointCloudLayer({
...
 
-* Note that although UTM ([Universal Transverse Mercator](https://en.wikipedia.org/wiki/Universal_Transverse_Mercator_coordinate_system)) coordinates uses similar meter offsets as the deck.gl meters mode, be aware there are subtle differences, so be careful before making assumptions.
+##### Screen space
 
+Top-left coordinate system runs from `[0, 0]` to `[viewportWidth, viewportHeight]`, measured in pixels.
```
Comment (kylebarron) [inline docs/developer-guide/coordinate-systems.md:46]:
```suggestion
A top-left coordinate system that runs from `[0, 0]` to `[viewportWidth, viewportHeight]`, measured in pixels.
```

File: docs/developer-guide/coordinate-systems.md [hunk 7]
```diff
@@ -19,42 +17,69 @@ new PointCloudLayer({
...
+| `COORDINATE_SYSTEM.CARTESIAN`         | `[x, y, z]` | `[x, y, z]` (optional, default `[0, 0, 0]`) | A linear system that measures equally on all 3 axes. |
+
+Remarks:
+
+* Although [Universal Transverse Mercator](https://en.wikipedia.org/wiki/Universal_Transverse_Mercator_coordinate_system) uses similar notions as the `METER_OFFSETS` mode, be aware that the deck.gl offset system does not have the sophistication of the UTM spec and should not be used interchangeably. See the [limitations](#limitations-of-the-offset-system) section for details.
+* The `CARTESIAN` mode describes positions that are identical in the world space and the common space. It is the default coordinate system when rendering into non-geopatial views. When combined with geospatial views, the positions are treated as common space coordinates for that particular projection mode. The latter can be seen used by the [MVTLayer](/docs/api-reference/geo-layers/mvt-layer.md), where the data decoded from the tiles are already pre-projected onto the Web Mercator plane.
```
Comment (kylebarron) [inline docs/developer-guide/coordinate-systems.md:72]:
```suggestion
* The `CARTESIAN` mode describes positions that are identical in the world space and the common space. It is the default coordinate system when rendering into non-geospatial views. When combined with geospatial views, the positions are treated as common space coordinates for that particular projection mode. The latter can be seen used by the [MVTLayer](/docs/api-reference/geo-layers/mvt-layer.md), where the data decoded from the tiles are already pre-projected onto the Web Mercator plane.
```

**Review by kylebarron — APPROVED** [review]
Looks great!

File: docs/api-reference/core/layer.md (hunk 1 above)
Comment (ibgreen) [reply docs/api-reference/core/layer.md:244]: Wouldn't hurt listing the modes?

**Review by ibgreen — APPROVED** [review]
Looks good. 

Not for this PR but at some point we might want to make a pass on the docs and more clearly emphasize the difference between data coordinate systems and projections. In particular if we want to gradually improve support for non-web-mercator data and non-web-mercator projections.

**Review by igorDykhta — APPROVED** [review]

**Pull request merged**